cmd/cli.go

```diff
@@ -77,6 +84,65 @@ var dumpCmd = &cobra.Command{
 	},
 }
 
+var containerdDumpCmd = &cobra.Command{
+	Use:   "containerd",
+	Short: "Manually checkpoint a running container to a directory",
+	Args:  cobra.ArbitraryArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cli, err := NewCLI()
+		if err != nil {
+			return err
+		}
+
+		a := api.ContainerDumpArgs{
+			Ref:         ref,
+			ContainerId: containerId,
+		}
+
+		var resp api.ContainerDumpResp
+		err = cli.conn.Call("CedanaDaemon.ContainerDump", a, &resp)
+		if err != nil {
+			return err
+		}
+		cli.logger.Info().Msgf("container %s dumped successfully to %s", containerId, dir)
+		return nil
+	},
+}
+
+var runcDumpCmd = &cobra.Command{
+	Use:   "runc",
+	Short: "Manually checkpoint a running container to a directory",
+	Args:  cobra.ArbitraryArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cli, err := NewCLI()
+		if err != nil {
+			return err
+		}
+
+		criuOpts := &container.CriuOpts{
+			ImagesDirectory: runcPath,
+			WorkDirectory:   workPath,
+			LeaveRunning:    true,
+			TcpEstablished:  false,
+		}
+
+		a := api.RuncDumpArgs{
+			PID:      runcPid,
+			WorkPath: workPath,
+			RuncPath: runcPath,
+			CriuOpts: *criuOpts,
```

Comment:
I think we could use the CriuOpts made in prepareDump here, would be a great first step towards unifying all our checkpointing

Reply:
Good point

Reply:
Can I turn this into a ticket? And proceed for now